### PR TITLE
Unit Testing

### PR DESCRIPTION
**Pull request by tin0d0**
- add unit testing capabilites
- add tests for primary cosntructors and getters

### CODE REVIEW

**Comment by tin0d0** [comment]
All the commits besides the last commit came from document PR, idk why so disregard those

File: project/src/main/java/com/google/sps/Models/Database.java [hunk 1]
```diff
@@ -51,24 +51,26 @@ public static User logInUser(String email, String nickname) {
     Entity userEntity = getDatastore().prepare(query).asSingleEntity();
 
     if(userEntity != null) {
-      return new User(email, nickname, userEntity.getKey().getId());
+      ArrayList<String> docHashes = (ArrayList)userEntity.getProperty("docHashes");
```
Comment (g-luo) [inline project/src/main/java/com/google/sps/Models/Database.java:54]: Are these changed in the document branch as well? If so, I think we should remove changes to Database, User, and Document to keep things separate.

Comment (g-luo) [reply project/src/main/java/com/google/sps/Models/Database.java:54]: If we don't need the changes Database, etc. we can push straight to master

**Review by g-luo — APPROVED** [review]
lgtm

**Review by max-friedman — APPROVED** [review]
Next time try to keep the Database and Document changes on their own branch. Just make sure that those are also reflected there. Other than that the testing code LGTM.

**Pull request closed without merging**